pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -573,12 +574,13 @@ private void processStreamEvents(MessageBatch messagesAndOffsets, long idlePipeS
       streamMessageCount++;
     }
     updateCurrentDocumentCountMetrics();
-    if (streamMessageCount != 0) {
```

Comment:
Does the behavior consistent for other streaming solutions like kinesis?

Reply:
Also will this behavior change metrics reporting

Reply:
Yes, this part is shared across all streams.
The unfiltered message count is always >= message count, so the existing behavior is preserved